src/cpp/src/continuous_batching_pipeline.cpp

```diff
@@ -217,6 +241,60 @@ class ContinuousBatchingPipeline::Impl {
             }
         }
 
+#ifdef DEBUG_CACHE_STATE_DUMP
+        static size_t step_count = 0;
+
+        CacheStateDumper dumper(CacheStateDumper::get_run_id_for_generation_step(step_count, "before_eviction"));
+        dumper.dump_cache_state(*m_scheduler, m_requests);
+#endif
+        const auto& sched_config = m_scheduler->get_config();
+
+
+        // evict unimportant blocks from KV cache, if requested
+        if (sched_config.use_cache_eviction) {
+            std::unordered_map<SequenceGroup::Ptr, size_t> seq_group_to_num_blocks_evicted_map;
```

Comment:
can we move this cache eviction logic outside of `step` function and just keep high-level call here?
`step()` is supposed to be high level function consisting of other blocks to keep readability.

Reply:
Done